ocp_resources/virtual_machine_instance.py

```diff
@@ -115,16 +115,20 @@ def wait_until_running(self, timeout=TIMEOUT_4MINUTES, logs=True, stop_status=No
             TimeoutExpiredError: If VMI failed to run.
         """
         try:
-            self.logger.info(
-                f"VMI {self.name} status before wait: {self.instance.status.phase}"
-            )
+            if self.exists:
+                self.logger.info(
+                    f"VMI {self.name} status before wait: {self.instance.status.phase}"
+                )
             self.wait_for_status(
                 status=self.Status.RUNNING, timeout=timeout, stop_status=stop_status
             )
         except TimeoutExpiredError:
             if not logs:
                 raise
-            self.logger.error(f"VMI {self.name} status: {self.instance.status.phase}")
+            if self.exists:
+                self.logger.error(
+                    f"VMI {self.name} status: {self.instance.status.phase}"
+                )
```

Comment:
Should we make an else and log that vmi doesn't exist?

Reply:
The user should start the vm (if it was started is should be existed) so no, no need.

Why `if self.exists:` needed here?

Reply:
If the user started the VM, and VMI wasn't created - it's a bug and it would be nice to log this behavior. 
But @vsibirsk already checked that in this case, we'll not get to this line, and will fail on a `NotFoundError` in `wait_for_status`

Reply:
calling `self.instance` or `self.virt_launcher_pod` raise kubernetis api exception (404 not found in case VMI doesn't exist)
updated the except block here so that it handles this correctly and raises `TimeoutExpiredError`

Reply:
Then `ocp_resources/virtual_machine.py:105` needs to be modified with an option to check if the vmi exists

Reply:
`VirtualMachine` is not involved here
here `self.wait_for_status`is called, not self.wait_for_ready_status (`VirtualMachineInstance` inherits from `NamespacedResource`, not from `VIrtualMachine`)
modifying `Resource.wait_for_status()` is not needed since it's not what is causing the issue

Reply:
> VirtualMachine is not involved here

It does (it is not just addressing the issue) - the right flow is to start a VM, wait for it to be ready, and **optionally** check `vmi.exists` and continue the flow.

Reply:
correct, but all of this is done outside of `VIrtualMachineInstance` methods (and VM won't ever be ready, if VMI is not ready)
this PR is needed just to handle the **unexpected** exception
i don't thinks it is a right way to call a class method and get a big traceback if something goes wrong

Reply:
it makes sense for `try` block. but in `except` block there's no point because we already hit timeout when vmi is not created